ocaml/src/albamgr_plugin.ml

```diff
@@ -291,13 +311,38 @@ let get_namespace_osds
      let state = deserialize Protocol.Osd.NamespaceLink.from_buffer (KV.cur_get_value cur) in
      (osd_id, state))
 
-let add_work_items work_items =
-  [ Log_plugin.make_update_x64
-      ~next_id_key:Keys.Work.next_id
-      ~log_prefix:Keys.Work.prefix
-      ~msgs:(List.map
-               (serialize Protocol.Work.to_buffer)
-               work_items) ]
+let add_work_items (db : read_user_db) ?(check=true) work_items =
+  let module W = Protocol.Work in
+  let secondary,_ =
+    List.fold_left
+      (fun (acc,names) item ->
+        match item with
+        | W.IterNamespace (W.Verify _, nsid, name, cnt) ->
+          let key = Keys.Work.job_name_prefix ^ name in
+          if StringSet.mem name names || db # exists key
+          then
+            let open Protocol in
+            Error.failwith
+              ~payload:(Printf.sprintf "job with name %S exists" name)
+              Error.Bad_argument
```

Comment:
this check should be done in the log plugin too? (otherwise you might have a race...)